src/test/java/uk/gov/hmcts/reform/coh/controller/RelistingControllerTest.java

```diff
@@ -0,0 +1,153 @@
+        mockMvc.perform(get(pathToNonExistingOnlineHearing).contentType(APPLICATION_JSON))
+            .andExpect(status().isNotFound())
+            .andExpect(content().string("Not found"));
+    }
+
+    @Test
+    public void creatingNewRelistForNonExistingOnlineHearingReturns404() throws Exception {
+        Relisting obj = new Relisting("Foo bar", RelistingState.DRAFTED);
+        String request = objectMapper.writeValueAsString(obj);
+        mockMvc.perform(post(pathToNonExistingOnlineHearing).content(request).contentType(APPLICATION_JSON))
+            .andExpect(status().isNotFound())
+            .andExpect(content().string("Not found"));
+    }
+
+    @Test
+    public void returns400WhenCreatingWithInvalidState() throws Exception {
+        Relisting obj = new Relisting("Foo bar", RelistingState.DRAFTED);
+        String request = new ObjectMapper().writeValueAsString(obj).replaceAll("DRAFTED", "SMELLED");
+        mockMvc.perform(post(pathToExistingOnlineHearing).content(request).contentType(APPLICATION_JSON))
+            .andExpect(status().isBadRequest());
+    }
+
+    @Test
+    public void cannotUpdateReasonAfterIssuing() throws Exception {
+        Relisting first = new Relisting("Final reason", RelistingState.ISSUED);
+        String req1 = objectMapper.writeValueAsString(first);
+        mockMvc.perform(post(pathToExistingOnlineHearing).content(req1).contentType(APPLICATION_JSON))
+            .andExpect(status().isAccepted());
+
+        Relisting second = new Relisting("This is the ultimate final reason!", RelistingState.ISSUED);
+        String req2 = objectMapper.writeValueAsString(second);
+        mockMvc.perform(post(pathToExistingOnlineHearing).content(req2).contentType(APPLICATION_JSON))
+            .andExpect(status().isConflict())
+            .andExpect(content().string("Already issued"));
+
+        mockMvc.perform(get(pathToExistingOnlineHearing).contentType(APPLICATION_JSON))
+            .andExpect(status().isOk())
+            .andExpect(jsonPath("$.reason", is(first.reason)))
+            .andExpect(jsonPath("$.state", is(first.state.toString())));
+    }
+
+    @Test
+    public void relistingStateIsCaseInsensitive() throws Exception {
+        String request = "{\"reason\":\"Test\",\"state\":\"drAfTed\"}";
+        mockMvc.perform(post(pathToExistingOnlineHearing).content(request).contentType(APPLICATION_JSON))
+            .andExpect(status().isAccepted());
+
+        mockMvc.perform(get(pathToExistingOnlineHearing).contentType(APPLICATION_JSON))
+            .andExpect(status().isOk())
+            .andExpect(jsonPath("$.reason", is("Test")))
+            .andExpect(jsonPath("$.state", is("DRAFTED")));
+    }
+
+    @Test
+    public void creatingRelistingRequiresStateField() throws Exception {
+        String request = "{}";
+        mockMvc.perform(post(pathToExistingOnlineHearing).content(request).contentType(APPLICATION_JSON))
+            .andExpect(status().isUnprocessableEntity());
+    }
+}
```

Comment:
Write Cucumber tests

Reply:
Thanks, I'll add some in the next commits.